dapr/proto/scheduler/v1/scheduler.proto

```diff
@@ -2,47 +2,39 @@ syntax = "proto3";
 
 package dapr.proto.scheduler.v1;
 
+import "google/protobuf/any.proto";
 import "google/protobuf/empty.proto";
 import "dapr/proto/runtime/v1/dapr.proto";
 
 option go_package = "github.com/dapr/dapr/pkg/proto/scheduler/v1;scheduler";
 
 service Scheduler {
-	// ConnectHost is used by the daprd sidecar to connect to the scheduler service.
-	rpc ConnectHost(ConnectHostRequest) returns (ConnectHostResponse) {}
 	// ScheduleJob is used by the daprd sidecar to schedule a job.
 	rpc ScheduleJob(ScheduleJobRequest) returns (ScheduleJobResponse) {}
 	// Get a job
 	rpc GetJob(GetJobRequest) returns (GetJobResponse) {}
 	// DeleteJob is used by the daprd sidecar to delete a job.
 	rpc DeleteJob(DeleteJobRequest) returns (DeleteJobResponse) {}
+  // WatchJob is used by the daprd sidecar to connect to the Scheduler
+  // service to watch for jobs triggering back.
+  rpc WatchJob(StreamJobRequest) returns (stream StreamJobResponse) {}
 }
 
-message ConnectHostRequest {
-	// The id of the application (app_id).
+message StreamJobRequest {
+	// The app_id of the sidecar.
 	string app_id = 1;
-	// The namespace.
+	// The namespace of the sidecar.
 	string namespace = 2;
-	// The hostname of the daprd sidecar.
-	string hostname = 3;
-	// The port of the daprd sidecar.
-	int32 port = 4;
 }
 
-message ConnectHostResponse {
-	// Empty as of now
-}
+message StreamJobResponse {
+	// Job data.
+	google.protobuf.Any data = 1;
 
-// UnregisterRequest is the message used by the daprd sidecar to unregister itself from the scheduler service.
-message UnregisterRequest {
-	// The id of the application (app_id).
-	string app_id = 1;
-	// The namespace.
-	string namespace = 2;
-	// The hostname of the daprd sidecar.
-	string hostname = 3;
-	// The port of the daprd sidecar.
-	int32 port = 4;
+	// The metadata associated with the job.
+	// The sidecar will create the unique `key` for storing data in the state store and pass the generated `key` along to the scheduler service for data lookup upon ‘trigger’ time later on.
+	// The sidecar will also add metadata in order to know whether this job is registered for an actor. This is needed, as the routing mechanism for actors is different for the callback.
+	map<string,string> metadata = 2;
```

Comment:
We should have a `Job` message type in this proto package. It is an internal API, different to the runtime (application) API.

Reply:
That's fine by me, but I'd like to do that in a separate PR if possible since its not part of my changes in this PR.

Reply:
I added a code todo for this